include/boost/url/impl/url.ipp

```diff
@@ -2271,63 +2274,64 @@ void
 url::
 build_tab() noexcept
 {
-#if 0
-    // path
-    if(nseg_ > 1)
+    BOOST_ASSERT(s_);
+
+    if (nseg_ != 0)
```

Comment:
we already know the beginning of the first segment (`offset(id_path)`) and the end of the last segment (`offset(id_query)`). So the table doesn't need to store these.

Reply:
Yes. I saw that logic for fields. 

Are they also always at the beginning for segments and query parameters? 

I thought these views (segments and params) implemented more complex logic to get rid of edge case prefixes and suffixes so I thought it wasn't worth reimplementing this logic elsewhere.

Reply:
I don't think there are any edge cases, but I could be wrong. Its up to you.

Reply:
There are some considerations that make it a little more complex than fields because the first element is not at index `0`.

Like

https://github.com/CPPAlliance/url/blob/fbc51ed52412d1421767effa92951ff417c1691f/include/boost/url/detail/path.hpp#L23

and

https://github.com/CPPAlliance/url/blob/fbc51ed52412d1421767effa92951ff417c1691f/include/boost/url/impl/url_view.ipp#L402

This logic is not complex. It's just more paths to cover because inferring the first `n` elements requires us to replicate this logic elsewhere to reconstruct these views to find the element.

But I'll start with the simpler logic and adapt on top of that.